sql/src/test/java/com/rilldata/calcite/CalciteToolboxTest.java

```diff
@@ -0,0 +1,123 @@
+package com.rilldata.calcite;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.apache.calcite.sql.dialect.PostgresqlSqlDialect;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+public class CalciteToolboxTest
+{
+  @Test
+  public void testMigrations() throws SqlParseException, JsonProcessingException
+  {
+    List<MigrationStep> migrationSteps = CalciteToolbox.inferMigrations("create view a as select 1", """
+          {
+            "entities": [
+              {
+                "name": "b",
+                "columns": [
+                  {
+                    "name": "a",
+                    "type": "int"
+                  }
+                ],
+                "ddl": "create table b (a int)"
+              }
+            ]
+          }
+          """, PostgresqlSqlDialect.DEFAULT);
+    Assertions.assertEquals("DROP TABLE \"B\"", migrationSteps.get(0).ddl);
+    Assertions.assertEquals("CREATE VIEW \"A\" AS\nSELECT 1", migrationSteps.get(1).ddl);
+    Assertions.assertEquals(2, migrationSteps.size());
+  }
+
+  @Test
+  public void testMigrationsWithArtifacts() throws SqlParseException, JsonProcessingException
+  {
+    List<MigrationStep> migrationSteps = CalciteToolbox.inferMigrations(
+        """
+            create view a as select 'a' as d, 2 as m ;
+            create metrics view b dimensions d measures count(m) from a
+        """,
+        """
+          {
+            "entities": [
+              {
+                "name": "b",
+                "columns": [
+                  {
+                    "name": "a",
+                    "type": "int"
+                  }
+                ],
+                "ddl": "create table b (a int)"
+              }
+            ]
+          }
+        """, PostgresqlSqlDialect.DEFAULT);
+    Assertions.assertEquals("DROP TABLE \"B\"", migrationSteps.get(0).ddl);
+    Assertions.assertEquals("CREATE VIEW \"A\" AS\nSELECT 'a' AS \"D\", 2 AS \"M\"", migrationSteps.get(1).ddl);
```

Comment:
what about the ```create metrics view b``` how is it handled here ?

Reply:
it should create an artefact creation step

Reply:
fixed